search/localsearch_nqueens.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * @file
+ * @brief [Heuristic local Search](https://en.wikipedia.org/wiki/Local_search_(optimization)) algorithm.
+ * @details
+ * The LS algorithm is the basis of many methods used for optimization problems.
+ * Its system is based on the search for neighborhoods that obtain a better solution.
+ * This exercise solves the n queens problem [https://en.wikipedia.org/wiki/Eight_queens_puzzle].
+ * The rule of neighboord applied in this file is first.improvement.
+ * @author [dhernandezl](https://github.com/dhernandezl)
+ */
+
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <array>
+
+/**
+ * function: initial_solution
+ * Initialize a new solution, generating
+ * random numbers with the maximum height of the board.
+ * @returns a vector that will be the starting solution.
+ */
+std::vector<int> initial_solution(int nqueens) {
+	std::vector<int> solution {};
+	solution.resize(nqueens);
+	for(auto& element: solution){
+		element = rand() % nqueens + 1;
+	}
+	return solution;
+}
+
+/**
+ * function: cost
+ * Calculate the number of pairs of queens that are attacked within the board,
+ * the optimal cost is h = 1. Read the 8 queens problem.
+ * @param [in] solution_vector is a vector with the
+ * column number where the queens are located on the board.
+ * @returns the cost of the solution vector.
+ */
+int cost(const std::vector<int> solution_vector) {
+	int h = 0;
+	//sum column
+	for (size_t i = 0; i < solution_vector.size(); i++){
+		for (size_t j = i + 1; j < solution_vector.size(); j++){
+			if (solution_vector[i] == solution_vector[j])
+			{
+				h++;
+				break;
+			}
+		}
+	}
+	//sum diagonal
+	for (size_t i = 0; i < solution_vector.size(); i++){
+		for (size_t j = i + 1; j < solution_vector.size(); j++){
+			if ((solution_vector[i] - (i + 1)) == (solution_vector[j] - (j + 1)))
+			{
+				h++;
+				break;
+			}
+		}
+	}
+	//sum inverse diagonal
+	for (size_t i = 0; i < solution_vector.size(); i++){
+		for (size_t j = i + 1; j < solution_vector.size(); j++) {
+			if ((solution_vector[i] + (i + 1)) == (solution_vector[j] + (j + 1))) {
+				h++;
+				break;
+			}
+		}
+	}
+	return h;
+}
+
+/**
+ * function: neighbour
+ * Generate a neighbor based on the current solution of
+ * the problem. Read about neighborhoods in local search.
+ * @param [in] solution_vector is a vector with the
+ * column number where the queens are located on the board.
+ * @returns a new vector similar to the one passed by parameter.
+ */
+std::vector<int> neighbour(std::vector<int> solution_vector) {
+	long i = rand() % solution_vector.size() + 1;
+	long mov = (rand() % solution_vector.size() + 1);

```

Comment:
Don't use the C-type `long`.
Use the `int64_t` type instead.
```suggestion
	int64_t i = rand() % solution_vector.size() + 1;
	int64_t mov = (rand() % solution_vector.size() + 1);
```

Reply:
Done¡ 
Thank you¡

Reply:
If you wanted to publish a simple genetic algorithm of discrete representation, how could you and in which directory?